lib/Parser/Differentiation.pm

```diff
@@ -12,28 +12,26 @@
 #
 sub Parser::D {
 	my $self = shift;
-	my $d;
-	my @x = @_;
-	my $x;
-	if (defined($x[0]) && $x[0] =~ m/^\d+$/) {
-		$d = shift(@x);
-		$self->Error("You can only specify one variable when you give a derivative count")
-			unless scalar(@x) <= 1;
-		return ($self) if $d == 0;
-	}
-	if (scalar(@x) == 0) {
-		my @vars = keys(%{ $self->{variables} });
-		my $n    = scalar(@vars);
-		if ($n == 0) {
-			return $self->new('0') if $self->{isNumber};
-			$x = 'x';
-		} else {
-			$self->Error("You must specify a variable to differentiate by") unless $n == 1;
-			$x = $vars[0];
+	my @vars = keys %{ $self->{variables} };
+	my $X    = @vars == 1 ? $vars[0] : undef;
+	push(@_, $X) if @_ == 0 && defined $X;
```

Comment:
I think that this should be `CORE::push(@_, $X) if @_ == 0 && defined $X;`, and line 32 should use `CORE::push` as well.

I am seeing the warnings `Ambiguous call resolved as CORE::push(), qualify as such or use &` from these lines  when running the unit tests.  Perl wants it clarified that this is not the `Parser::push` method.

Reply:
I will fix it.